src/integ-test/java/com/mathworks/ci/RunMATLABTestsInteg.java

```diff
@@ -0,0 +1,263 @@
+        this.buildWrapper.setMatlabRootFolder(getMatlabroot());
+        project.getBuildWrappersList().add(this.buildWrapper);
+        project.getBuildersList().add(this.testBuilder);
+        HtmlPage page = jenkins.createWebClient().goTo("job/test0/configure");
+        HtmlCheckBoxInput modelCoverageChkBx = page.getElementByName("modelCoverageArtifact");
+        modelCoverageChkBx.setChecked(true);
+        Thread.sleep(2000);
+        HtmlTextInput coberturaModelCoverageFileInput=(HtmlTextInput) page.getElementByName("_.modelCoverageFilePath");
+        Assert.assertEquals(TestData.getPropValues("modelcoverage.file.path"),coberturaModelCoverageFileInput.getValueAttribute());
+    }
+
+
+    @Test
+    public void verifySTMResultsFilePathInput() throws Exception {
+        this.buildWrapper.setMatlabRootFolder(getMatlabroot());
+        project.getBuildWrappersList().add(this.buildWrapper);
+        project.getBuildersList().add(this.testBuilder);
+        HtmlPage page = jenkins.createWebClient().goTo("job/test0/configure");
+        HtmlCheckBoxInput stmResultsChkBx = page.getElementByName("stmResultsArtifact");
+        stmResultsChkBx.setChecked(true);
+        Thread.sleep(2000);
+        HtmlTextInput STMRFilePathInput=(HtmlTextInput) page.getElementByName("_.stmResultsFilePath");
+        Assert.assertEquals(TestData.getPropValues("stmresults.file.path"),STMRFilePathInput.getValueAttribute());
+    }
+
+    /*
+     * Test to verify if Matrix build fails when MATLAB is not available.
+     */
+    @Test
+    public void verifyMatrixBuildFails() throws Exception {
+        MatrixProject matrixProject = jenkins.createProject(MatrixProject.class);
+        Axis axes = new Axis("VERSION", "R2018a", "R2018b");
+        matrixProject.setAxes(new AxisList(axes));
+        String matlabRoot = getMatlabroot();
+        this.buildWrapper.setMatlabRootFolder(matlabRoot.replace("R2018b", "$VERSION"));
+        matrixProject.getBuildWrappersList().add(this.buildWrapper);
+
+
+        matrixProject.getBuildersList().add(testBuilder);
+
+        // Check for first matrix combination.
+
+        Map<String, String> vals = new HashMap<String, String>();
+        vals.put("VERSION", "R2018a");
+        Combination c1 = new Combination(vals);
+        MatrixRun build1 = matrixProject.scheduleBuild2(0).get().getRun(c1);
+
+
+        jenkins.assertBuildStatus(Result.FAILURE, build1);
+
+        // Check for second Matrix combination
+
+        Combination c2 = new Combination(vals);
+        MatrixRun build2 = matrixProject.scheduleBuild2(0).get().getRun(c2);
+
+        jenkins.assertBuildStatus(Result.FAILURE, build2);
+    }
+    
+    /*
+     * Test to verify if Matrix build passes (mock MATLAB).
+     */
+    @Test
+    public void verifyMatrixBuildPasses() throws Exception {
+        MatrixProject matrixProject = jenkins.createProject(MatrixProject.class);
+        Axis axes = new Axis("VERSION", "R2020a", "R2020a");
```

Comment:
Are the two versions mentioned here supposed to be same?

Reply:
No, the versions need not be same but they have to be installed on the machine. I will correct the test case with different versions.